.mergify.yml

```diff
@@ -37,7 +37,46 @@ pull_request_rules:
         # Choices are either merge base branch into PR (default) or rebase the PR against the base branch
         strict_method: rebase
 
-  - name: Automatic merge for release branch
+  - name: Automatic squash to main branch
+    conditions:
+      # True if the base branch is main
+      - base=main
+      # check-success are true if the check ran successfully
+      - "check-success=Shellcheck"
+      - "check-success=Lint"
+      - "check-success=Tidy"
+      - "check-success=Mocks"
+      - "check-success=Chart checks"
+      - "check-success=Go build"
+      - "check-success=Go test"
+      - "check-success=Test various Envoy + SMI configuration scenarios"
+      - "check-success=Integration Test with Tresor, SMI traffic policies, and egress disabled"
+      - "check-success=Go test e2e (1)"
+      - "check-success=Go test e2e (2)"
+      - "check-success=Go test e2e (3)"
+      - "check-success=Go test e2e (4)"
+      - "check-success=DCO"
+      - "check-success=netlify/osm-docs/deploy-preview"
+      # True if the PR has at least 2 approving reviews
+      - "#approved-reviews-by>=2"
+      # True if the PR has the automerge label
+      - label=automerge-squash
```

Comment:
Does this check if the 2 approvals are from core maintainers? ref: https://github.com/openservicemesh/osm/blob/main/CONTRIBUTING.md#merging-pull-requests

Reply:
@SanyaKochhar I think that is enforced by the Github setting on the repo, don't think mergify can bypass that.